packages/core/src/VNodes/VNode.ts

```diff
@@ -544,14 +544,12 @@ export class VNode {
     descendants<T>(predicate?: Predicate<T>): VNode[];
     descendants<T>(predicate?: Predicate<T>): VNode[] {
         const descendants = [];
-        if (this.children.length) {
-            let currentDescendant = this.firstChild();
-            do {
-                if (currentDescendant.test(predicate)) {
-                    descendants.push(currentDescendant);
-                }
-                currentDescendant = this._descendantAfter(currentDescendant);
-            } while (currentDescendant);
+        let currentDescendant = this.firstChild();
```

Comment:
Isn't this a REF instead of a FIX?

Reply:
Well, I just switched from a `do...while` to a `while` and removed the `if` that was useless. Somewhat a bit of both in some way but the reason I did it was to fix a bug. (and the true "refactor" part is just removing a useless if)